pkg/odo/cli/logs/logs.go

```diff
@@ -89,6 +90,18 @@ func (o *LogsOptions) Complete(ctx context.Context, cmdline cmdline.Cmdline, _ [
 }
 
 func (o *LogsOptions) Validate(ctx context.Context) error {
+
+	switch fcontext.GetPlatform(ctx, commonflags.PlatformCluster) {
+	case commonflags.PlatformCluster:
+		if o.clientset.KubernetesClient == nil {
+			return errors.New("you need access to a cluster to run this command")
+		}
+	case commonflags.PlatformPodman:
+		if o.clientset.PodmanClient == nil {
+			return errors.New("you need access to podman to run this command")
```

Comment:
Can you rebase this on top of main and use the new `kclient.NewNoConnectionError()` function?
Same for podman, `podman.NewPodmanNotFoundError()`.